spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java

```diff
@@ -0,0 +1,56 @@
+package org.springframework.cloud.dataflow.rest.client.config;
+
+import java.net.URI;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.dataflow.core.DataFlowPropertyKeys;
+import org.springframework.cloud.dataflow.rest.client.DataFlowOperations;
+import org.springframework.cloud.dataflow.rest.client.DataFlowTemplate;
+import org.springframework.cloud.dataflow.rest.client.dsl.Stream;
+import org.springframework.cloud.dataflow.rest.client.dsl.StreamBuilder;
+import org.springframework.cloud.dataflow.rest.util.HttpClientConfigurer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Configuration class to provide default beans for {@link DataFlowOperations} and {@link org.springframework.cloud.dataflow.rest.client.dsl.StreamBuilder} instances.
+ * @author Vinicius Carvalho
+ */
+@Configuration
+@EnableConfigurationProperties(DataFlowClientProperties.class)
+@ConditionalOnProperty(prefix = DataFlowPropertyKeys.PREFIX + "client", name = "enableDsl")
```

Comment:
If you have the client library on your classpath, then you should get the `DataFlowTemplate` and `StreamBuilder`.  This is similar to how other `xyzTemplate` classes are created via autoconfiguration, e.g. RabbitTemplate, RedisTemplate etc.

Reply:
We discussed this before. This is due shell, it causes conflict with it's created template. I believe you and I get in an agreement that we should use conditional properties because of that (instead of changing the shell code to figure out which template to use), but if that has changed I can work on changing the shell classes instead.

Reply:
Changing the shell class to adapt to autoconfig would be the way to go.